### PR TITLE
changed: avoid AddonManager.h includes in headers

### PR DESCRIPTION
**Pull request by notspiff**
and remove unnecessary includes

## Description
Reduce hotness of AddonManager.h
Before: 134 files rebuilt
After: 84 files rebuilt

## Motivation and context
Less rebuilds

## How has this been tested?
It builds

## What is the effect on users?
None

## Types of change

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [x] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:

- [x] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

File: xbmc/ContextMenuManager.h [hunk 1]
```diff
@@ -63,10 +68,10 @@ namespace CONTEXTMENU
   /*!
    * Starts the context menu loop for a file item.
    * */
-  bool ShowFor(const CFileItemPtr& fileItem, const CContextMenuItem& root=CContextMenuManager::MAIN);
+  bool ShowFor(const std::shared_ptr<CFileItem>& fileItem, const CContextMenuItem& root=CContextMenuManager::MAIN);
```
Comment (ksooo) [inline xbmc/ContextMenuManager.h:71]: Please add `#include <memory>`to this header.

File: xbmc/ContextMenus.h [hunk 2]
```diff
@@ -18,22 +18,22 @@ struct CEjectDisk : CStaticContextMenuAction
 {
   CEjectDisk() : CStaticContextMenuAction(13391) {} // Eject/Load CD/DVD!
   bool IsVisible(const CFileItem& item) const override;
-  bool Execute(const CFileItemPtr& item) const override;
+  bool Execute(const std::shared_ptr<CFileItem>& item) const override;
```
Comment (ksooo) [inline xbmc/ContextMenus.h:21]: Please add `#include <memory>`to this header.

File: xbmc/favourites/ContextMenus.h [hunk 3]
```diff
@@ -20,34 +20,34 @@ class CFavouriteContextMenuAction : public CStaticContextMenuAction
 public:
   explicit CFavouriteContextMenuAction(uint32_t label) : CStaticContextMenuAction(label) {}
   bool IsVisible(const CFileItem& item) const override;
-  bool Execute(const CFileItemPtr& item) const override;
+  bool Execute(const std::shared_ptr<CFileItem>& item) const override;
```
Comment (ksooo) [inline xbmc/favourites/ContextMenus.h:23]: Please add `#include <memory>`to this header.

File: xbmc/music/ContextMenus.h [hunk 4]
```diff
@@ -18,7 +18,7 @@ struct CMusicInfo : CStaticContextMenuAction
 {
   explicit CMusicInfo(MediaType mediaType);
   bool IsVisible(const CFileItem& item) const override;
-  bool Execute(const CFileItemPtr& item) const override;
+  bool Execute(const std::shared_ptr<CFileItem>& item) const override;
```
Comment (ksooo) [inline xbmc/music/ContextMenus.h:21]: Please add `#include <memory>`to this header.

File: xbmc/video/ContextMenus.h [hunk 5]
```diff
@@ -20,7 +20,7 @@ class CVideoInfo : public CStaticContextMenuAction
 public:
   explicit CVideoInfo(MediaType mediaType);
   bool IsVisible(const CFileItem& item) const override;
-  bool Execute(const CFileItemPtr& item) const override;
+  bool Execute(const std::shared_ptr<CFileItem>& item) const override;
```
Comment (ksooo) [inline xbmc/video/ContextMenus.h:23]: Please add `#include <memory>`to this header.

**Comment by ksooo** [comment]
Maybe xbmc/ContextMenuItem.h can also be optimized?

File: xbmc/ContextMenuManager.h (hunk 1 above)
Comment (notspiff) [reply xbmc/ContextMenuManager.h:71]: done

File: xbmc/ContextMenus.h (hunk 2 above)
Comment (notspiff) [reply xbmc/ContextMenus.h:22]: done

File: xbmc/favourites/ContextMenus.h (hunk 3 above)
Comment (notspiff) [reply xbmc/favourites/ContextMenus.h:25]: done

File: xbmc/video/ContextMenus.h (hunk 5 above)
Comment (notspiff) [reply xbmc/video/ContextMenus.h:25]: done

**Comment by notspiff** [comment]
i did some stuff there, not sure if that's what you had in mind.

**Comment by ksooo** [comment]
Yes, that’s what I had in mind. 👍

File: xbmc/music/ContextMenus.h (hunk 4 above)
Comment (notspiff) [reply xbmc/music/ContextMenus.h:23]: done

**Review by ksooo — APPROVED** [review]
Nice cleanup.

**Review by garbear — APPROVED** [review]
LGTM

**Pull request merged**